packages/flutter_tools/test/general.shard/resident_runner_test.dart

```diff
@@ -423,6 +423,11 @@ void main() {
     expect(testLogger.errorText, contains('Error'));
   }));
 
+  test('ResidentTunner clears the screen when it should', () => testbed.run(() async {
+    residentRunner.clearScreen();
```

Comment:
Does this need to write something to the logger first? I'm not sure what terminal.clear screen actually does in tests...

Reply:
I wasn't sure either, but the test output clears and it's marked as "passed", so I guess it works.

Reply:
It sounds like it is clearing the actual test runner screen? That is probably not what we want to do. In this case, I would instead inject a mock Stdio and assert that the correct method is called

Reply:
It's a bit odd behavior, (at least on Windows, ) it "clears" the test screen, but really it's just pushed up to the top of the screen -- you can scroll right back up to see it again. 

As for mocking, `testLogger` is provided by this code (`test/lib/src/context.dart`)

```dart
/// Return the test logger. This assumes that the current Logger is a BufferLogger.
BufferLogger get testLogger => context.get<Logger>() as BufferLogger;
```

So, I changed `clearScreen` from: 

```dart
// Clears the screen.
void clearScreen() => globals.stdio.stdoutWrite(globals.terminal.clearScreen() + '\n');
```
to:
 
```dart
// Clears the screen.
void clearScreen([BufferLogger logger]) => logger == null
  ? globals.stdio.stdoutWrite(globals.terminal.clearScreen() + '\n')
  : logger.clear();
```

And then changed the tests to a test that looks like this, and it passed:
```dart
residentRunner.clearScreen(testLogger);
expect(testLogger.statusText, equals(''));
```